File: backend/src/main/java/com/condos/pago/dto/CuotaUsuarioResponse.java
package com.condos.pago.dto;

import com.condos.pago.model.EstadoPago;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
public class CuotaUsuarioResponse {
    private Long id;
    private Long cuotaId;
    private String concepto;
    private BigDecimal monto;
    private LocalDate fechaVencimiento;
    private Long usuarioId;
    private String usuarioNombre;
    private String unidadHabitacional;
    private EstadoPago estado;
    private String referenciaPago;
    private String notasUsuario;
    private String notasAdmin;
    private LocalDateTime fechaReporte;
    private LocalDateTime fechaConfirmacion;
}

File: lib/features/pagos/models/cuota_usuario_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'cuota_usuario_response.g.dart';

enum EstadoPago {
  @JsonValue('PENDIENTE')
  pendiente,

  @JsonValue('REPORTADO')
  reportado,

  @JsonValue('CONFIRMADO')
  confirmado,

  @JsonValue('RECHAZADO')
  rechazado,
}

@JsonSerializable()
class CuotaUsuarioResponse {
  final int id;
  final int cuotaId;
  final String concepto;
  final double monto;
  final String fechaVencimiento;
  final int usuarioId;
  final String usuarioNombre;
  final String? unidadHabitacional;
  final EstadoPago estado;
  final String? referenciaPago;
  final String? notasUsuario;
  final String? notasAdmin;
  final DateTime? fechaReporte;
  final DateTime? fechaConfirmacion;

  CuotaUsuarioResponse({
    required this.id,
    required this.cuotaId,
    required this.concepto,
    required this.monto,
    required this.fechaVencimiento,
    required this.usuarioId,
    required this.usuarioNombre,
    this.unidadHabitacional,
    required this.estado,
    this.referenciaPago,
    this.notasUsuario,
    this.notasAdmin,
    this.fechaReporte,
    this.fechaConfirmacion,
  });

  factory CuotaUsuarioResponse.fromJson(Map<String, dynamic> json) =>
      _$CuotaUsuarioResponseFromJson(json);
  Map<String, dynamic> toJson() => _$CuotaUsuarioResponseToJson(this);
}
